scipy/sparse/linalg/_interface.py

```diff
@@ -178,8 +178,15 @@ def _init_dtype(self):
         """Called from subclasses at the end of the __init__ routine.
         """
         if self.dtype is None:
-            v = np.zeros(self.shape[-1])
+            v = np.zeros(self.shape[-1], dtype=np.int8)
```

Comment:
Leave a comment here explaining why we use `int8`, otherwise, it's likely a later developer will change it.

Reply:
added to the function docstring above